### DISCUSSION TITLE
SE_Group 04_Diabetes Prediction System

### DISCUSSION THREAD

**Discussion by VenumiV** [Mathanraj-Sharma/intro-to-ml-sjp#19] [General]
Hello Sir,

We are Group 04 Neural Net-Nuts from the Software Engineering program, and this is our official discussion thread for the project Diabetes Prediction System.

Team Members:
- FC212029 W M Venumi V Alagoda 
- FC212039 Thusharangi Fernando
- FC212022 A A S Thathsara Bandara 
- FC212015 Mohammed Shimas Saburdeen 
- FC212002 Janith Nanayakkara 
- FC212026 Malsha Jayamanne

### GitHub Repo
https://github.com/VenumiV/diabetes-predictor-ml-flask.git

### Proposal Document
[ML_Project_Proposal.pdf](https://github.com/user-attachments/files/20790140/ML_Project_Proposal.pdf)

**Comment by Mathanraj-Sharma** [comment]
@VenumiV 

I expect the following from each team member:

- Each member should train a model using a different machine learning algorithm than their peers.
- Each member is responsible for conducting independent exploratory data analysis (EDA) and data preprocessing, tailored to the algorithm they are using.
- Hyperparameter tuning should be performed to optimize each model’s performance.
- Everyone must contribute to testing and evaluation of their own models.
- Among all trained models, the best-performing model should be selected and deployed in an application.

All training code should be committed to the team’s code repository. In the final presentation, a clear model comparison must be provided, with justification for how and why the winning model was chosen.

**Comment by Mathanraj-Sharma** [comment]
@VenumiV Weekly Evaluation for your Team

-  No actual development in the project, just project structure available. 
- Remove `.Trash` folder. it is from someone's local file system. Ignore them from being committed to the project repo by adding in .gitignore

Weekly Score: 1/10

**Comment by Mathanraj-Sharma** [comment]
@VenumiV Weekly Evaluation 

- No EDA and jumped to model training
    - https://github.com/VenumiV/diabetes-predictor-ml-flask/blob/main/notebooks/FC110595_thathsara/XGBoot.ipynb
- Why are you normalizing features here
    - https://github.com/VenumiV/diabetes-predictor-ml-flask/blob/main/notebooks/FC212002_Janith/descisiontree.ipynb
- Can someone explain what is interpreted from this missing value heatmap 
    - https://github.com/VenumiV/diabetes-predictor-ml-flask/blob/main/notebooks/FC212029_Venumi/RandomForest.ipynb
- I can see only see works from 4 students
- Good progress compare to last week

Weekly Score: 6/10
